charts/cni-metrics-helper/README.md

```diff
@@ -0,0 +1,64 @@
+# CNI METRICS HELPER
+
+This chart provides a Kubernetes deployment for the Amazon VPC CNI Metrics Helper, which is used to collect metrics for the Amazon VPC CNI plugin for Kubernetes.
+
+## Prerequisites
+
+- Kubernetes 1.11+ running on AWS
+- Helm 3.0+
+
+## Installing the Chart
+
+Clone the Amazon VPC CNI for Kubernetes repository to your local machine.
+
+```shell
+$ git clone https://github.com/aws/amazon-vpc-cni-k8s.git
+```
+Use the helm install command to install the chart into your Kubernetes cluster
+
+```shell
+$ helm install cni-metrics-helper ./amazon-vpc-cni-k8s/charts/cni-metrics-helper
+```
+
+## Configuration
+
+The following table lists the configurable parameters for this chart and their default values.
+
+| Parameter                  | Description                                                   | Default            |
+|----------------------------|---------------------------------------------------------------|--------------------|
+| fullnameOverride           | Override the fullname of the chart                            | cni-metrics-helper |
+| image.region               | ECR repository region to use. Should match your cluster       | us-west-2          |
+| image.tag                  | Image tag                                                     | v1.11.4            |
+| image.account              | ECR repository account number                                 | 602401143452       |
+| image.domain               | ECR repository domain                                         | amazonaws.com      |
+| env.USE_CLOUDWATCH         | Whether to export CNI metrics to CloudWatch                   | true               |
+| env.AWS_CLUSTER_ID         | ID of the cluster to use when exporting metrics to CloudWatch | default            |
+| metricUpdateInterval       | Interval at which to update CloudWatch metrics, in seconds    | 60                 |
+| serviceAccount.name        | The name of the ServiceAccount to use                         | nil                |
+| serviceAccount.create      | Specifies whether a ServiceAccount should be created          | true               |
+| serviceAccount.annotations | Specifies the annotations for ServiceAccount                  | {}                 |
+
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install` or provide a YAML file containing the values for the above parameters:
+
+```shell
+$ helm install my-release ./amazon-vpc-cni-k8s/charts/cni-metrics-helper --set useCloudwatch=false --values values.yaml
+
+```
+
+## Resources
```

Comment:
For this to work properly, you will need to add a `resources` section to `values.yaml` in the existing helm chart. Note that the existing deployment already handles when resources are set: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/charts/cni-metrics-helper/templates/deployment.yaml#L23

So the steps would be:
1. Copy resources from https://github.com/aws/amazon-vpc-cni-k8s/blob/master/charts/aws-vpc-cni/values.yaml#L116
2. Increment helm chart version: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/charts/cni-metrics-helper/Chart.yaml#L18
3. Update README to capture `resources` defaults

Reply:
About the 3rd point , since we have set the defaults for cpu only in the resources section ( 25m ) So i need to have only this parameter? You can check the latest commit and let me know about it.

Reply:
Good call out, so technically we just need the top-level `resources` field to be present, and then anyone can override it. I think we should follow what our integration tests do and just define it as an empty dictionary: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/test/helm/charts/cni-metrics-helper/values.yaml#L41

Very strange that this was missing in the first place. But yeah, then we can just add the `resources` entry to the README like:
```
| `resources`             | Resources for the pods                                  | `{}`                 |
```
then we don't have to worry about defaults

Reply:
Should i also remove the example ? or is it correct?

Reply:
Yeah, let's update
```
resources:
  requests:
    cpu: 25m
```
in `values.yaml` to
```
resources: {}
```
and then I think this is good to go! Since this is a user-facing change, can you also update the release note in the PR description to something like "added resources field to cni-metrics-helper chart"

Thanks for all of the help!

Reply:
Done!